src/util/psbt/serialize.rs

```diff
@@ -70,7 +71,20 @@ impl Deserialize for PublicKey {
     }
 }
 
-impl Serialize for KeySource {
+impl Serialize for ExtendedPubKey {
+    fn serialize(&self) -> Vec<u8> {
+        self.encode().to_vec()
+    }
+}
+
+impl Deserialize for ExtendedPubKey {
+    fn deserialize(bytes: &[u8]) -> Result<Self, Error> {
+        ExtendedPubKey::decode(&bytes)
+            .map_err(|_| Error::ParseFailed("xpub parse error"))
+    }
+}
+
+impl Serialize for (Fingerprint, DerivationPath) {
```

Comment:
Why was this changed back from using `KeySource` to the composite type?

Reply:
This is a mistake, will change. The code was just based on an older version before that PR with `KeySource` was merged